proto/storage/cluster.proto

```diff
@@ -94,6 +95,43 @@ message AdmissionControllerConfig {
   bool enforce_on_updates = 5;
 }
 
+message ProcessConfig {
+  bool enabled = 1;
+}
+
+message NetworkConnectionConfig {
+  bool enabled = 1;
+  bool aggregate_external = 2;
+}
+
+message NetworkEndpointConfig {
+  bool enabled = 1;
+  bool include_listening_endpoint_processes = 2;
+}
+
+message CollectorFeature {
+  oneof feature {
+    ProcessConfig processes = 1;
+    NetworkConnectionConfig network_connections = 2;
+    NetworkEndpointConfig network_endpoints = 3;
+  }
+}
+
+message CollectorNamespaceConfig {
+  message NamespaceRule {
+    string namespace = 1;
+    storage.MatchType match_type = 2;
+  }
+
+  CollectorFeature feature = 1;
+  repeated NamespaceRule namespace_selection = 2;
+}
+
+message CollectorConfig {
+  repeated CollectorFeature cluster_scope_config = 1;
+  repeated CollectorNamespaceConfig namespace_scope_config = 2;
```

Comment:
Could this be directly part of `CollectorFeature`? 
In this case I think you would need to resolve the `CollectorNamespaceConfig` to a `CollectorFeature`.

Reply:
`CollectorFeature` can be one of `ProcessConfig`, `NetworkConnectionConfig`, or `NetworkEndpointConfig`. `CollectorNamespaceConfig` contains a field of type `CollectorFeature`, so including `CollectorNamespaceConfig` in `CollectorFeature` results in recursion. Perhaps `CollectorFeature` could be better named?

The idea here that for the cluster level configuration all that is needed is one instance of `ProcessConfig`, `NetworkConnectionConfig`, and `NetworkEndpointConfig`, while for the namespace level config a more complicated structure is needed.

Reply:
If only one instance of each of the configs is necessary, why not hardcode them in the collector config?

```
message CollectorConfig {
  ProcessConfig process_config = 1;
  NetworkConnectionConfig network_connection_config = 2;
  ...
}
```

Reply:
Done